src/Messaging/Handlers/RetailerCreatedHandler.cs

```diff
@@ -0,0 +1,52 @@
+﻿namespace Linn.DemStock.Messaging.Handlers
+{
+    using System.Text;
+
+    using Linn.Common.Logging;
+    using Linn.Common.Messaging.RabbitMQ;
+    using Linn.Common.Messaging.RabbitMQ.Unicast;
+    using Linn.Common.Persistence;
+    using Linn.DemStock.Domain;
+    using Linn.DemStock.Domain.Repositories;
+    using Linn.DemStock.Resources.External;
+
+    using Newtonsoft.Json;
+
+    public class RetailerCreatedHandler
+    {
+        private readonly ITransactionManager transactionManager;
+
+        private readonly ILog log;
+
+        private readonly IRabbitTerminator rabbitTerminator;
+
+        private readonly IRetailerDemListRepository retailerDemListRepository;
+
+        public RetailerCreatedHandler(
+            ITransactionManager transactionManager,
+            ILog log,
+            IRabbitTerminator rabbitTerminator,
+            IRetailerDemListRepository retailerDemListRepository)
+        {
+            this.transactionManager = transactionManager;
+            this.log = log;
+            this.rabbitTerminator = rabbitTerminator;
+            this.retailerDemListRepository = retailerDemListRepository;
+        }
+
+        public bool Execute(IReceivedMessage message)
+        {
+            var content = Encoding.UTF8.GetString(message.Body);
+            var resource = JsonConvert.DeserializeObject<RetailerResource>(content);
+            this.log.Info($"Creating dem list for retailer {resource.Id} {resource.Name}");
+
+            var factory = new RetailerDemListCreator(this.retailerDemListRepository);
+            factory.CreateRetailerDemList(resource.Id, "/employees/100");
```

Comment:
What employee record is this, and why are we hardcoding it? If things are created automatically surely they should have a null employee URI?

Reply:
100 = automated transaction

Reply:
Huh, didn't know we had a fake employee for that purpose. Might be worth putting a comment in explaining that.